ci-operator/step-registry/gcp/provision/static-creds/gcp-provision-static-creds-commands.sh

```diff
@@ -0,0 +1,213 @@
+# user specified image pullspec, to avoid auth error when accessing it, always use build farm 
+# registry pullspec.
+echo "RELEASE_IMAGE_LATEST_FROM_BUILD_FARM: ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}"
+# seem like release-controller does not expose RELEASE_IMAGE_INITIAL, even job configuraiton defines 
+# release:initial image, once that, use 'oc get istag release:inital' to workaround it.
+echo "RELEASE_IMAGE_INITIAL: ${RELEASE_IMAGE_INITIAL:-}"
+if [[ -n ${RELEASE_IMAGE_INITIAL:-} ]]; then
+  tmp_release_image_initial=${RELEASE_IMAGE_INITIAL}
+  echo "Getting inital release image from RELEASE_IMAGE_INITIAL..."
+elif oc get istag "release:initial" -n ${NAMESPACE} &>/dev/null; then
+  tmp_release_image_initial=$(oc -n ${NAMESPACE} get istag "release:initial" -o jsonpath='{.tag.from.name}')
+  echo "Getting inital release image from build farm imagestream: ${tmp_release_image_initial}"
+fi
+# For some ci upgrade job (stable N -> nightly N+1), RELEASE_IMAGE_INITIAL and 
+# RELEASE_IMAGE_LATEST are pointed to different imgaes, RELEASE_IMAGE_INITIAL has 
+# higher priority than RELEASE_IMAGE_LATEST
+TESTING_RELEASE_IMAGE=""
+if [[ -n ${tmp_release_image_initial:-} ]]; then
+  TESTING_RELEASE_IMAGE=${tmp_release_image_initial}
+else
+  TESTING_RELEASE_IMAGE=${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}
+fi
+echo "TESTING_RELEASE_IMAGE: ${TESTING_RELEASE_IMAGE}"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+GCP_SERVICE_ACCOUNT=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${GCP_SERVICE_ACCOUNT}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+# For OCP 4.15+
+CCO_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_cco_permissions"
+IMAGE_REGISTRY_CUSTOM_ROLE="projects/${GOOGLE_PROJECT_ID}/roles/installer_qe_image_registry_permissions"
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+MPREFIX="${SHARED_DIR}/manifest"
+creds_requests_dir="$(mktemp -d)"
+creds_keys_dir="$(mktemp -d)"
+creds_manifests_dir="$(mktemp -d)"
+
+echo "$(date -u --rfc-3339=seconds) - Extracting GCP credentials requests from the release image..."
+ADDITIONAL_OC_EXTRACT_ARGS=""
+if [[ "${EXTRACT_MANIFEST_INCLUDED}" == "true" ]]; then
+  ADDITIONAL_OC_EXTRACT_ARGS="${ADDITIONAL_OC_EXTRACT_ARGS} --included --install-config=${SHARED_DIR}/install-config.yaml"
+fi
+echo "OC Version:"
+which oc
+oc version --client
+oc adm release extract --help
+
+dir=$(mktemp -d)
+pushd "${dir}"
+cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+oc registry login --to pull-secret
+oc adm release extract --registry-config pull-secret --credentials-requests --cloud=gcp --to="${creds_requests_dir}" ${ADDITIONAL_OC_EXTRACT_ARGS} "${TESTING_RELEASE_IMAGE}"
+rm pull-secret
+popd
+
+if [ X"${FEATURE_SET}" != X"TechPreviewNoUpgrade" ]; then
```

Comment:
We can refer to the conditions in https://github.com/openshift/release/blob/master/ci-operator/step-registry/aws/provision/cco-manual-users/static/aws-provision-cco-manual-users-static-commands.sh#L151